docs/connectors/completion-tokens.md

```diff
@@ -0,0 +1,152 @@
+# Synchronous Processing with Completion Tokens
+
+Completion tokens enable synchronous processing on top of Feldera’s asynchronous pipelines, allowing
+users to precisely track when a specific set of inputs has been fully processed by the pipeline.
+
+## Background
+
+Feldera pipelines process data **asynchronously**. While the incremental SQL engine is evaluating
+earlier inputs, input connectors continue ingesting data into internal buffers, and output connectors
+write prior results to external sinks.
+
+![Architecture of a Feldera Pipeline](../pipelines/pipeline_architecture.png)
+
+However, the SQL engine itself is **synchronous** and **strongly consistent**. It processes inputs in
```

Comment:
you can point to the corresponding blog post https://www.feldera.com/blog/synchronous-streaming